pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -45,22 +44,48 @@ type ServiceBinderOptions struct {
 	Logger                 *log.Log
 	DynClient              dynamic.Interface
 	DetectBindingResources bool
-	EnvVarPrefix           string
 	SBR                    *v1alpha1.ServiceBindingRequest
 	Client                 client.Client
+	Objects                []*unstructured.Unstructured
+	EnvVars                map[string][]byte
+	EnvVarPrefix           string
+	Binding                *Binding
 }
 
-// Valid returns whether the options are valid.
-func (o *ServiceBinderOptions) Valid() bool {
-	return o.SBR != nil && o.DynClient != nil && o.Client != nil
+// ErrInvalidServiceBinderOptions is returned when ServiceBinderOptions contains an invalid value.
+type ErrInvalidServiceBinderOptions string
+
+func (e ErrInvalidServiceBinderOptions) Error() string {
+	return fmt.Sprintf("option %q is missing", string(e))
+}
+
+// Valid returns an error if the receiver is invalid, nil otherwise.
+func (o *ServiceBinderOptions) Valid() error {
```

Comment:
perhaps `IsValid`? It looks that the function is used only within this package, so it should be private?

Reply:
`IsValid` is as bad as `Valid` in this case, specially because it returns an `error`. Should be called `Validate` instead.